robusta_krr/core/abstract/strategies.py

```diff
@@ -61,21 +64,9 @@ def timeframe_timedelta(self) -> datetime.timedelta:
 ArrayNx2 = Annotated[NDArray[np.float64], Literal["N", 2]]
 
 
-class ResourceHistoryData(pd.BaseModel):
-    """A class to represent resource history data.
-
-    metric is the metric information used to gather the history data.
-    data is a mapping from pod to a numpy array of time and value.
-    """
-
-    metric: Metric
-    data: dict[str, ArrayNx2]  # Mapping: pod -> [(time, value)]
-
-    class Config:
-        arbitrary_types_allowed = True
+MetricPodData = dict[str, ArrayNx2]  # Mapping: pod -> [(time, value)]
+MetricsPodData = dict[type["BaseMetric"], MetricPodData]
```

Comment:
MetricsPodData and MetricPodData are sooooo simmilar, it confused me for a bit I would recommend renaming one of them to prevent future mixups.